doc/specs/#653 - Quake Mode/#653 - Quake Mode.md

```diff
@@ -0,0 +1,629 @@
+vs elevated windows.
+* `"whenElevated": null`: behave as normal - the first context level to run wins
+* `"whenElevated": true`: only register the hotkey when running elevated
+* `"whenElevated": false`: only register the hotkey when running unelevated
+
+</td>
+</tr>
+<tr>
+<td><strong>OneCore / Windows 10X</strong></td>
+<td>
+
+I'm fairly certain that none of these APIs would work on Windows 10X at all.
+These features would have to initially be disabled in a pure UWP version of the
+Terminal, until we could find workarounds. Since the window layer is the one
+responsible for the management of the hotkeys and the tray icon, we're not too
+worried about this.
+
+</td>
+</tr>
+</table>
+
+If there are any other applications running that have already registered hotkeys
+with `RegisterHotKey`, then it's possible that the Terminal's attempt to
+register that hotkey will fail. If that should happen, then we should display a
+warning dialog to the user indicating which hotkey will not work (because it's
+already used for something else).
+
+Which is the "current" monitor? The one with the mouse or the one with the
+active window? This isn't something that has an obvious answer. Guake implements
+this feature where the "current monitor" is the one with the mouse on it. At
+least for the first iterations of this action, that's what we'll use.
+
+`monitor: onCurrent|onCurrentWindow|toCurrent|<int>`
+
+
+## Implementation plan
+
+Currently, in [`dev/migrie/f/653-QUAKE-MODE`], I have some sample rudimentary
+code to implement quake mode support. It allows for only a single global hotkey
+that summons the MRU window, without dropdown. That would be a good place for
+anyone starting to work on this feature. From there, I imagine the following
+work would be needed:
+
+* [ ] Change the `globalHotKey` to a list of `globalHotkeys`. `AppHost` would
+  need to be able to get _all_ of these hotkeys, and register all of them. Each
+  one would need to be assigned a unique ID, so `WM_HOTKEY` can identify which
+  hotkey was pressed.
+    - This could be committed without any other args to the `globalHotkeys`.
+      We'd assume the "default" behavior or summoning the MRU window, where it
+      is, no dropdown, to start with. From there, we'd add the remaining
+      properties:
+    * [ ] Add support for the `hideWhenVisible` property
+    * [ ] Add support for the `desktop` property to control how window summoning
+      interacts with virtual desktops
+    * [ ] Add support for the `monitor` which monitor the window appears on.
+    * [ ] Add support for the `dropdownDuration` property
+* [ ] Add the `minimizeToTray` setting, and implement it without any sort of flyout
+    * [ ] Add a list of windows to the right-click flyout on the tray icon
+    * [ ] Add support for the `alwaysShowTrayIcon` setting
+
+
+### Future Considerations
+
+I don't believe there are any other tracked requests that are planned that
+aren't already included in this spec.

```

Comment:
Back in Quake 1 and Half-Life the console window would expand only on half of the screen.

Would this be supported? i.e. will there be a setting to specify height of the  terminal window once it slides down?

Reply:
Yea, a whole section was added to the _very specific_ quake mode, which is a special case of summoning the window.